File: tests/helpers_py/management_api.py
# SPDX-License-Identifier: Apache-2.0

"""Management API helpers for integration tests.

Mirrors tests/helpers/management-api.ts.
"""

from __future__ import annotations

import httpx

from .urls import MANAGEMENT_URL


async def resolve_test_org_id(token: str) -> str:
    """List the user's organizations and return the seeded ``test-org`` id."""
    async with httpx.AsyncClient(timeout=10.0) as client:
        resp = await client.get(
            f"{MANAGEMENT_URL}/organizations",
            headers={"Authorization": f"Bearer {token}"},
        )
        resp.raise_for_status()
        data = resp.json()

    for org in data.get("items", []):
        if org.get("slug") == "test-org":
            return org["id"]

    slugs = [o.get("slug") for o in data.get("items", [])]
    raise RuntimeError(
        f"Test org not found. Available: {slugs}. Run: pnpm run db:seed"
    )


def org_auth_headers(token: str, org_id: str) -> dict[str, str]:
    """Build auth + org headers for management API calls."""
    return {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {token}",
        "X-Organization-Id": org_id,
    }

File: tests/helpers/management-api.ts
// SPDX-License-Identifier: Apache-2.0

/**
 * Shared helpers for integration tests that call the management API.
 *
 * After org-scoping was added, all agent endpoints require an org context
 * (either from the JWT's organizationId claim or the X-Organization-Id header).
 * These helpers resolve the test org and build the correct headers so tests
 * work regardless of which org the JWT defaults to.
 */

import { MANAGEMENT_URL } from './urls';

/**
 * Resolve the test org ID by listing the user's organizations and finding
 * the seeded `test-org`.  Throws with a descriptive message if not found.
 */
export async function resolveTestOrgId(token: string): Promise<string> {
  const res = await fetch(`${MANAGEMENT_URL}/organizations`, {
    headers: { Authorization: `Bearer ${token}` },
  });
  if (!res.ok) throw new Error(`Failed to list orgs: ${res.status}`);
  const data = (await res.json()) as {
    items: { id: string; slug: string; isPersonal: boolean }[];
  };
  const testOrg = data.items.find((o) => o.slug === 'test-org');
  if (!testOrg) {
    throw new Error(
      `Test org not found. Available orgs: ${data.items.map((o) => o.slug).join(', ')}. Run: pnpm run db:seed`,
    );
  }
  return testOrg.id;
}

/**
 * Build auth + org headers for management API calls.
 */
export function orgAuthHeaders(
  token: string,
  orgId: string,
): Record<string, string> {
  return {
    'Content-Type': 'application/json',
    Authorization: `Bearer ${token}`,
    'X-Organization-Id': orgId,
  };
}
